src/metabase/api/api_key.clj

```diff
@@ -129,4 +129,19 @@
   (let [api-keys (t2/hydrate (t2/select :model/ApiKey) :group_name :updated_by)]
     (map present-api-key api-keys)))
 
+(api/defendpoint DELETE "/:id"
+  "Delete an ApiKey"
+  [id]
+  {id ms/PositiveInt}
+  (api/check-superuser)
+  (t2/with-transaction [_tx]
+    (let [api-key (-> (t2/select-one :model/ApiKey id)
+                      (t2/hydrate :group_name))]
```

Comment:
Not a huge deal, but might be worth wrapping this in `check-404`. Even though it's a deletion endpoint, we should probably return a 404 if the caller is trying to delete something that already doesn't exist. 

The User update call below would also error in this case anyway, so probably better to catch that at the start and error.

Reply:
Good call. I'm also going to add this check to the `PUT` endpoints.